lib/system/shared/mesh-client.js

```diff
@@ -221,7 +221,17 @@
 
     var client = {
       _mesh: {},
-      log: opts.log
+      log: opts.log,
+      updateSession:function(session){
+        var _this = this;
+        _this.session = session;
+        if (_this._mesh && _this._mesh.endpoints){
+          Object.keys(_this._mesh.endpoints).forEach(function(endpointName){
+              var endpoint = _this._mesh.endpoints[endpointName];
+              endpoint.messenger.session.id = session.id;
```

Comment:
why apply the same session ID to all endpoints? don't we know which endpoint we're connected to at this point?